### PR TITLE
bump k8s to 1.23

### PR DESCRIPTION
**Pull request by atiratree**
Depends on these bumps:
- [x]  openshift/api#1076
- [x] openshift/client-go#203
- [x] openshift/library-go#1259

### CODE REVIEW

File: cmd/cluster-kube-controller-manager-operator/main.go [hunk 1]
```diff
@@ -59,5 +45,7 @@ func NewSSCSCommand(ctx context.Context) *cobra.Command {
 	cmd.AddCommand(certsyncpod.NewCertSyncControllerCommand(operator.CertConfigMaps, operator.CertSecrets))
 	cmd.AddCommand(recoverycontroller.NewCertRecoveryControllerCommand(ctx))
 
+	globalflag.AddGlobalFlags(cmd.PersistentFlags(), cmd.Name(), logs.SkipLoggingConfigurationFlags())
```
Comment (atiratree) [inline cmd/cluster-kube-controller-manager-operator/main.go:48]: - non deprecates logging flags are present in global options (`PersistentFlags`) by default (see `cli.Run`). The following command omits them and keeps them only in each command that specifically needs them. It also helps with distinguishing exclusively new and deprecated options (present in global options) and clustering each group together.
- user experience can be also enhanced by adding `"logs"` flagset for the logging configuration (example https://github.com/kubernetes/kubernetes/blob/4df70dfd41601077478289ad358e7c10a4f0d0e1/cmd/kube-controller-manager/app/options/options.go#L276)

File: pkg/cmd/render/render.go [hunk 2]
```diff
@@ -5,25 +5,31 @@ import (
...
+
+	"github.com/openshift/cluster-kube-controller-manager-operator/bindata"
+	"github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/targetconfigcontroller"
 )
 
 // renderOpts holds values to drive the render command.
 type renderOpts struct {
 	manifest genericrenderoptions.ManifestOptions
 	generic  genericrenderoptions.GenericOptions
+	logs     *logs.Options
```
Comment (atiratree) [inline pkg/cmd/render/render.go:32]: each `Command` needs to have log options

File: pkg/cmd/render/render.go [hunk 3]
```diff
@@ -37,12 +43,19 @@ func NewRenderCommand(errOut io.Writer) *cobra.Command {
 	renderOpts := &renderOpts{
 		manifest: *genericrenderoptions.NewManifestOptions("kube-controller-manager", "openshift/origin-hyperkube:latest"),
 		generic:  *genericrenderoptions.NewGenericOptions(),
+		logs:     logs.NewOptions(),
```
Comment (atiratree) [inline pkg/cmd/render/render.go:46]: sets reasonable defaults (eg. `FlushFrequency`)

File: pkg/cmd/render/render.go [hunk 4]
```diff
@@ -37,12 +43,19 @@ func NewRenderCommand(errOut io.Writer) *cobra.Command {
...
 	}
 	cmd := &cobra.Command{
 		Use:   "render",
 		Short: "Render kubernetes controller manager bootstrap manifests, secrets and configMaps",
 		Run: func(cmd *cobra.Command, args []string) {
+			// Activate logging as soon as possible.
+			if err := renderOpts.logs.ValidateAndApply(); err != nil {
+				fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+				os.Exit(1)
+			}
```
Comment (atiratree) [inline pkg/cmd/render/render.go:57]: this validates the logging flags and initializes klog

File: pkg/cmd/render/render.go [hunk 5]
```diff
@@ -59,6 +72,7 @@ func NewRenderCommand(errOut io.Writer) *cobra.Command {
 	}
 
 	renderOpts.AddFlags(cmd.Flags())
+	renderOpts.logs.AddFlags(cmd.Flags())
```
Comment (atiratree) [inline pkg/cmd/render/render.go:75]: add the flags to the current command

**Review by atiratree — COMMENTED** [review]
~This is an explanation that serves as a template for other operators.~

This explanantion is useful only for custom solution of logs flags.

**Comment by atiratree** [comment]
^ https://github.com/openshift/cluster-kube-controller-manager-operator/pull/581#pullrequestreview-826829435 only applies for a custom solution of logs flags. 


Please see current code for a general solution

**Comment by atiratree** [comment]
removed the forks and updated

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**